localstack/services/cloudwatch/provider_v2.py

```diff
@@ -183,12 +185,14 @@ def delete_alarms(self, context: RequestContext, alarm_names: AlarmNames) -> Non
         """
         Delete alarms.
         """
-
-        for alarm_name in alarm_names:
-            alarm_arn = arns.cloudwatch_alarm_arn(
-                alarm_name, account_id=context.account_id, region_name=context.region
-            )  # obtain alarm ARN from alarm name
-            self.alarm_scheduler.delete_scheduler_for_alarm(alarm_arn)
+        with _STORE_LOCK:
+            for alarm_name in alarm_names:
+                alarm_arn = arns.cloudwatch_alarm_arn(
+                    alarm_name, account_id=context.account_id, region_name=context.region
+                )  # obtain alarm ARN from alarm name
+                self.alarm_scheduler.delete_scheduler_for_alarm(alarm_arn)
+                store = self.get_store(context.account_id, context.region)
+                store.Alarms.pop(alarm_arn, None)
```

Comment:
nit: unrelated to this PR, but the `store` attributes `Alarms` with a capital A does not look too pythonic... would there be a way to change it before the provider goes live and that it would break persistence? I believe it should be `alarms`. Was there a specific reason for it being Pascal cased?

Reply:
https://github.com/localstack/localstack/pull/10087